webroot/js/components/chat/chat-message-view.js

```diff
@@ -0,0 +1,46 @@
+import { h, Component } from '/js/web_modules/preact.js';
+import htm from '/js/web_modules/htm.js';
+const html = htm.bind(h);
+
+import { messageBubbleColorForString } from '../../utils/user-colors.js';
+import { formatMessageText, formatTimestamp } from '../../utils/chat.js';
+import { generateAvatar } from '../../utils/helpers.js';
+
+export default class ChatMessageView extends Component {
+  render() {
+    const { message, username } = this.props;
+    const { image, author, body, timestamp } = message;
+
+    const formattedMessage = formatMessageText(body, username);
+    const avatar = image || generateAvatar(author);
+    const formattedTimestamp = formatTimestamp(timestamp);
+
+    const authorColor = messageBubbleColorForString(author);
+    const avatarBgColor = { backgroundColor: authorColor };
+    const authorTextColor = { color: authorColor };
+
```

Comment:
Since it seems that only a few tailwind classes are different for the different types of message, you could make some consts here like 

```
const messageContentStyle = classNames({
 'message-content': true,
' text-sm': message.type === SYSTEM,
 'break-words': true,
  'w-full': true,
});
```
..then pass this to the message-content div, `div class=${messageContentStyle}..`
you may need to import classNames from utils.

Reply:
Ok I made the system message more different so they don't have to be combined.  I just don't want to limit what we do with system messages because we combine the style with non-system messages.  I want to be able to iterate on one without touching the other.

Reply:
you can probably still do it in one file.
you could also break out system message in to a sub-component of chat-message, so you don't have to repeat 70% of the code in this file.

Reply:
That's assuming the final look of this message will have 70% of the same code.  I think this change may be premature because I don't know what it should look like, and I don't want to make decisions now that limit us later, so I'm closing this until there's a real plan behind it.